src/rep.rs

```diff
@@ -5,6 +5,48 @@ use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum ImageBuildChunk {
+    Stream {
+        stream: String,
+    },
+    Error {
+        error: String,
+        #[serde(rename = "errorDetail")]
+        error_detail: ErrorDetail,
+    },
+    Aux {
+        aux: Aux,
+    },
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Aux {
+    #[serde(rename = "ID")]
+    id: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ErrorDetail {
+    message: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ImagePullChunk {
+    status: String,
+    id: Option<String>,
+    progress: Option<String>,
+    #[serde(rename = "progressDetail")]
+    progress_detail: Option<ProgressDetail>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ProgressDetail {
+    current: Option<u64>,
+    total: Option<u64>,
+}
```

Comment:
I feel like this would be better as an enum instead of a struct with optional fields, but I can't think of a good way to slice it...

Reply:
I don't think it makes sense to split it up into an enum as all fields can be present at the same time.